e2e_tests/conftest.py

```diff
@@ -65,17 +65,49 @@ def environment():
     )
 
 
+def initialize_container(container_type, project):
+    if container_type == "project":
+        return neptune.init_project(name=project)
+
+    if container_type == "run":
+        return neptune.init_run(project=project)
+
+    if container_type == "model":
+        return neptune.init_model(key=a_key(fake.slug()), project=project)
+
+    if container_type == "model_version":
+        model = neptune.init_model(key=a_key(fake.slug()), project=project)
+        model_sys_id = model["sys/id"].fetch()
+        model.stop()
+
+        return neptune.init_model_version(model=model_sys_id, project=project)
+
+
 @pytest.fixture(scope="session")
 def container(request, environment):
-    if request.param == "project":
-        project = neptune.init_project(name=environment.project)
-        yield project
-        project.stop()
-
-    if request.param == "run":
-        exp = neptune.init_run(project=environment.project)
-        yield exp
-        exp.stop()
+    container = initialize_container(
+        container_type=request.param, project=environment.project
+    )
+
+    yield container
+
+    container.stop()
+
+
+@pytest.fixture(scope="session")
+def containers_pair(request, environment):
+    container_a_type, container_b_type = request.param.split("-")
```

Comment:
I strongly believe you can pass a not-string (tuple) to `request.param` -- it's going to be cleaner

Reply:
Yes, you could but then PyCharm couldn't handle it in human-readable way :wink: Test name will be "testname-container0" instead of "testname-model-run"

Reply:
sold :)